packages/nuxt/src/runtime/utils.ts

```diff
@@ -78,3 +86,35 @@ export function reportNuxtError(options: {
     });
   });
 }
+
+async function flushWithTimeout(): Promise<void> {
+  const sentryClient = getClient();
+  const isDebug = sentryClient ? sentryClient.getOptions().debug : false;
+
+  try {
+    isDebug && logger.log('Flushing events...');
+    await flush(2000);
+    isDebug && logger.log('Done flushing events');
+  } catch (e) {
+    isDebug && logger.log('Error while flushing events:\n', e);
```

Comment:
these don't need `isDebug`, they should be doing the sentry client check under the hood
```suggestion
  try {
    isDebug && logger.log('Flushing events...');
    await flush(2000);
    isDebug && logger.log('Done flushing events');
  } catch (e) {
    isDebug && logger.log('Error while flushing events:\n', e);
```
```suggestion
  try {
    logger.log('Flushing events...');
    await flush(2000);
    logger.log('Done flushing events');
  } catch (e) {
    logger.log('Error while flushing events:\n', e);
```